eppo_core/src/ufc/assignment.rs

```diff
@@ -29,10 +31,38 @@ pub enum AssignmentValue {
     /// A boolean value.
     Boolean(bool),
     /// Arbitrary JSON value.
-    Json(Arc<serde_json::Value>),
+    Json {
+        raw: Str,
+        parsed: Arc<serde_json::Value>,
+    },
+}
+
+impl PartialEq for AssignmentValue {
```

Comment:
cool!

Reply:
haha, `PartialEq` was auto-generated before but with adding `raw` json strings, we can't compare on them (because serialization may differ), so I had to roll my own implementation — that's rather un-cool 😅